Installing custom requirements.txt must happen before plugin_paths.py

Thanks for merging #63, I just tested it with Solr, but unfortunately my implementation had a slight bug.

In our site, we import the python module `checksumdir` in [pelicanconf.py](https://github.com/apache/solr-site/blob/3a02aca42058f349267ca02c0e21cdce62d33795/pelicanconf.py#L5). This dependency is declared in [requirements.txt](https://github.com/apache/solr-site/blob/3a02aca42058f349267ca02c0e21cdce62d33795/requirements.txt#L3).

However, in the action script, the `pelican/plugin_paths.py` script is ran before we run pip install, so [then it fails](https://github.com/apache/solr-site/actions/runs/9661381496/job/26648970543?pr=108) with

```
Getting plugins from action location: /home/runner/work/_actions/apache/infrastructure-actions/main/pelican
Traceback (most recent call last):
  File "/home/runner/work/_actions/apache/infrastructure-actions/main/pelican/plugin_paths.py", line 67, in <module>
    main()
  File "/home/runner/work/_actions/apache/infrastructure-actions/main/pelican/plugin_paths.py", line 60, in main
    exec(contents, values) # pylint: disable=exec-used
  File "<string>", line 5, in <module>
ModuleNotFoundError: No module named 'checksumdir'
Error: Process completed with exit code 1.
```

The solution is to install requirements before parsing `pelicanconf.py`. Will create a PR.

--- a/pelican/action.yml
+++ b/pelican/action.yml
@@ -92,6 +92,13 @@ runs:
     - name: Generate website from markdown
       shell: bash
       run: |
+        if [ -f "${{ inputs.requirements }}" ]
+        then
+          echo "Installing python requirements from ${{ inputs.requirements }}"
+          pip3 install -r ${{ inputs.requirements }}
+        else
+          echo "Requirements file ${{ inputs.requirements }} does not exist"
+        fi
         if [ "${{ inputs.debug }}" == 'true' ]
         then
           OPTS='-D'
@@ -105,13 +112,6 @@ runs:
         echo "Getting plugins from action location: ${{ github.action_path }}"
         PP=$(python3 ${{ github.action_path }}/plugin_paths.py '${{ github.action_path }}/plugins')
         set -x # Show the expanded variables
-        if [ -f "${{ inputs.requirements }}" ]
-        then
-          echo "Installing python requirements from ${{ inputs.requirements }}"
-          pip3 install -r ${{ inputs.requirements }}
-        else
-          echo "Requirements file ${{ inputs.requirements }} does not exist"
-        fi
         python3 -B -m pelican content -e "$PP" -o ${{ inputs.tempdir }} $OPTS
 
     - name: Check out previous branch

